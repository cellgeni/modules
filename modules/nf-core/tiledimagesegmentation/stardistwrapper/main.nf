process TILEDIMAGESEGMENTATION_STARDISTWRAPPER {
    tag "$meta.id"
    label 'process_medium'

    conda "${moduleDir}/environment.yml"
    container "quay.io/cellgeni/tiled_stardist:0.9.1-0.1.0"

    input:
    tuple val(meta), val(x_min), val(y_min), val(x_max), val(y_max), path(image)

    output:
    tuple val(meta), path("${out_name}"), emit: wkts
    tuple val(meta), path("${stem}*png"), emit: cp_plots, optional: true
    path "versions.yml"           , emit: versions

    when:
    task.ext.when == null || task.ext.when

    script:
    stem = "${meta.id}-${x_min}_${y_min}_${x_max}_${y_max}"
    def args = task.ext.args ?: ''  
    out_name = "${stem}_cp_outlines.wkt"
    """
    export NUMBA_CACHE_DIR=/tmp/numba_cache
    stardist_helper.py run \
        --image_path ${image} \
        --x_min ${x_min} \
        --y_min ${y_min} \
        --x_max ${x_max} \
        --y_max ${y_max} \
        --output_name ${out_name} \
        ${args}
    
    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        tiledimagesegmentation/stardist: \$(stardist_helper.py version)
    END_VERSIONS
    """

    stub:
    def args = task.ext.args ?: ''
    stem = "${meta.id}-${x_min}_${y_min}_${x_max}_${y_max}"
    out_name = "${stem}_cp_outlines.wkt"
    """
    touch ${out_name}

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        tiledimagesegmentation/stardist: \$(stardist_helper.py version)
    END_VERSIONS
    """
}
