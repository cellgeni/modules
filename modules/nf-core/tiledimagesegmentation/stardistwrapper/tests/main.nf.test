// nf-core modules test tiledimagesegmentation/stardistwrapper
nextflow_process {

    name "Test Process TILEDIMAGESEGMENTATION_STARDISTWRAPPER"
    script "../main.nf"
    process "TILEDIMAGESEGMENTATION_STARDISTWRAPPER"
    config "./nextflow.config"

    tag "modules"
    tag "modules_nfcore"
    tag "tiledimagesegmentation"
    tag "tiledimagesegmentation/stardistwrapper"

    test("mouse_heart - tiff") {

        when {
            process {
                """
                input[0] = [
                    [ id:'mouse_heart'],
                    0, 0, 200, 200,
                    file(params.modules_testdata_base_path + 'imaging/tiff/mindagap.mouse_heart.wga.tiff', checkIfExists: true),
                ]
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }

    }

    test("mouse_heart - tiff - 500") {

        when {
            process {
                """
                input[0] = [
                    [ id:'mouse_heart'],
                    0, 0, 500, 500,
                    file(params.modules_testdata_base_path + 'imaging/tiff/mindagap.mouse_heart.wga.tiff', checkIfExists: true),
                ]
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }

    }

    test("mouse_heart - tiff - stub") {

        options "-stub"

        when {
            process {
                """
                input[0] = [
                    [ id:'mouse_heart'],
                    0, 0, 200, 200,
                    file(params.modules_testdata_base_path + 'imaging/tiff/mindagap.mouse_heart.wga.tiff', checkIfExists: true),
                ]
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }

    }
}
