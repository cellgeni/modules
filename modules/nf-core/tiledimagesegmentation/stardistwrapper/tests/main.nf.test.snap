{
    "mouse_heart - tiff - stub": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "mouse_heart"
                        },
                        "mouse_heart-0_0_200_200_cp_outlines.wkt:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "1": [
                    
                ],
                "2": [
                    "versions.yml:md5,fd0243e4f499988b39a40759f26f9e71"
                ],
                "cp_plots": [
                    
                ],
                "versions": [
                    "versions.yml:md5,fd0243e4f499988b39a40759f26f9e71"
                ],
                "wkts": [
                    [
                        {
                            "id": "mouse_heart"
                        },
                        "mouse_heart-0_0_200_200_cp_outlines.wkt:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.5"
        },
        "timestamp": "2025-05-06T23:01:10.197408867"
    },
    "mouse_heart - tiff": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "mouse_heart"
                        },
                        "mouse_heart-0_0_200_200_cp_outlines.wkt:md5,76d1d4913d3a581edbac889a672f42f3"
                    ]
                ],
                "1": [
                    
                ],
                "2": [
                    "versions.yml:md5,fd0243e4f499988b39a40759f26f9e71"
                ],
                "cp_plots": [
                    
                ],
                "versions": [
                    "versions.yml:md5,fd0243e4f499988b39a40759f26f9e71"
                ],
                "wkts": [
                    [
                        {
                            "id": "mouse_heart"
                        },
                        "mouse_heart-0_0_200_200_cp_outlines.wkt:md5,76d1d4913d3a581edbac889a672f42f3"
                    ]
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.5"
        },
        "timestamp": "2025-05-06T23:00:36.584861913"
    },
    "mouse_heart - tiff - 500": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "mouse_heart"
                        },
                        "mouse_heart-0_0_500_500_cp_outlines.wkt:md5,c8bc6acc0c45b483ca1f390d61871769"
                    ]
                ],
                "1": [
                    
                ],
                "2": [
                    "versions.yml:md5,fd0243e4f499988b39a40759f26f9e71"
                ],
                "cp_plots": [
                    
                ],
                "versions": [
                    "versions.yml:md5,fd0243e4f499988b39a40759f26f9e71"
                ],
                "wkts": [
                    [
                        {
                            "id": "mouse_heart"
                        },
                        "mouse_heart-0_0_500_500_cp_outlines.wkt:md5,c8bc6acc0c45b483ca1f390d61871769"
                    ]
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.5"
        },
        "timestamp": "2025-05-06T23:00:57.120009418"
    }
}